app/Http/Controllers/Teams/LeaveTeamController.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace App\Http\Controllers\Teams;
+
+use App\Http\Controllers\Controller;
+use App\Models\Teams\Team;
+use App\Models\User\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+
+class LeaveTeamController extends Controller
+{
+    public function __construct()
+    {
+        $this->middleware('auth');
+    }
+
+    public function __invoke(Request $request)
+    {
+        $request->validate([
+            'teamId' => 'required|exists:teams,id'
+        ]);
+
+        /** @var User $user */
+        $user = auth()->user();
+        /** @var Team $team */
+        $team = Team::find($request->teamId);
+
+        if (!$user->teams()->whereTeamId($request->teamId)->exists()) {
+            abort(403, 'You are not part of this team!');
```

Comment:
I would prefer to `return ['success' => false, 'msg' => '']` and handle it with a popup / toast notification

Reply:
These abort operations are meant to handle cases where users hackishly try to make requests. So they should never appear from users doing normal things. That's why we exit with an exception (aborting).